### PR TITLE
guess field names for map

### PR DESCRIPTION
**Pull request by christophsturm**
based on #88

this is a very defensive version of field name guessing, if it does not work it just defaults to the old behavior. 

the focus here is not to support any block syntax that anybody could ever write, but to have a great way to navigate to sub properties, and to replace property references. I think it works well enough to remove property references and use this syntax primarily, preferred without `it` for better readability. 

next step would be to generalize the method parsing to work for expect() too.

### CODE REVIEW

File: strikt-core/src/main/kotlin/strikt/api/Assertion.kt [hunk 1]
```diff
@@ -181,7 +183,15 @@ interface Assertion {
           "value of ${function.propertyName}",
           function
         )
-        else -> map("%s", function)
+        else -> {
+          val fieldName = try {
+            val line = FilePeek.getCallerFileInfo().line
+            ParsedMapInstruction(line).body.substringAfter("it.")
```
Comment (robfletcher) [inline strikt-core/src/main/kotlin/strikt/api/Assertion.kt]: Would this still work if the lambda parameter had an explicit name?

Comment (christophsturm) [reply strikt-core/src/main/kotlin/strikt/api/Assertion.kt]: no. in that case the block would fail and the default would trigger.

Comment (christophsturm) [reply strikt-core/src/main/kotlin/strikt/api/Assertion.kt]: actually i tried it and if i change it to 
`          map { sub->sub.birthDate.year }.isEqualTo(1971)`
the error looks like this:
`sub->sub.birthDate.year:`

Comment (christophsturm) [reply strikt-core/src/main/kotlin/strikt/api/Assertion.kt]: it just take the whole body of the map block, which is useful most of the time

Comment (robfletcher) [reply strikt-core/src/main/kotlin/strikt/api/Assertion.kt]: That's not bad, yeah

**Pull request merged**